planner-api/src/main/java/pl/edu/agh/kuce/planner/goal/service/GoalService.java

```diff
@@ -127,4 +126,17 @@ private List<SubGoalData> transformSubGoals(final List<SubGoal> subGoalList) {
                           .map(SubGoalData::new)
                           .toList();
     }
+
+    @Transactional
+    public GoalData completeSubGoal(final Integer subGoalId, final Integer goalId, final User user) {
+        final Optional<Goal> goal = goalRepository.getGoalById(goalId, user);
+        if (goal.isPresent()) {
+            final Optional<SubGoal> subGoal = subGoalRepository.getSubGoalById(subGoalId, user);
+            if (subGoal.isPresent()) {
+                subGoalRepository.completeSubGoal(subGoalId, goal.get());
```

Comment:
All the methot besides this line is the same as destroySubGoal. Consider extracting a new method. In addidtion, method destroy also has duplicate part (maybe 2 new methods?)